ch06ex02/test/LoginTest.js

```diff
@@ -38,40 +43,47 @@ describe("Login", function () {
     await driver.get("https://the-internet.herokuapp.com/login");
     await driver.findElement({ id: "username" }).sendKeys(USERNAME);
     // NOTE: We're appending some characters to the password here to give us an invalid password to test the messaging.
-    await driver.findElement({ id: "password" }).sendKeys(PASSWORD + "pants");
+    await driver.findElement({ id: "password" }).sendKeys(PASSWORD + Math.random());
     await driver.findElement({ css: "button" }).click();
 
-    assert(
-      await driver.findElement({ css: ".flash.error" }).isDisplayed(),
-      "Success message not displayed"
-    );
+    // NOTE: Another suboptimal issue might be the fact we're doing the DOM
+    // lookup of `driver.findElement({css:".flash.error"})` at least 3x in
+    // this test case. Might be worth seeing if we can do it once and cache
+    // the potentially expensive lookup or if that has downstream issues.
```

Comment:
This comment just moved up from the bottom of the file so it's more "in-context" with the code.